### PR TITLE
o/devicestate: check for previously installed revisions when falling back to installed snaps during offline remodel

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This PR enables offline remodeling to look back at recently installed revisions of snaps, and will use one of those if any fulfill the revision requirements of the remodel.

### CODE REVIEW

File: overlord/devicestate/devicestate.go [hunk 1]
```diff
@@ -531,6 +531,23 @@ func (ro *remodelVariant) UpdateWithDeviceContext(st *state.State, snapName stri
...
+		var ss snapstate.SnapState
+		if err := snapstate.Get(st, snapName, &ss); err != nil {
+			return nil, err
+		}
+
+		// if the current revision isn't the revision that is installed,
+		// then look at the previous revisions that we have to see if any of
+		// those match
+		for _, rev := range ss.Sequence.Revisions {
+			if rev.Snap.Revision == opts.Revision {
```
Comment (pedronis) [inline overlord/devicestate/devicestate.go:543]: this could use SnapSequence.LastIndex

**Review by pedronis — COMMENTED** [review]
thanks, one small comment. This can't be backported to 2.61 though as is

**Comment by pedronis** [comment]
this will be in 2.62 but not in 2.61.2

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

**Review by Meulengracht — APPROVED** [review]
Thank you

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**